app/models/stocktake.rb

```diff
@@ -58,29 +58,26 @@ class Stocktake < ApplicationRecord
   # @return [Array<StocktakeRevision>] The newly created revisions
   #
   def populate_revisions!
-    package_ids = PackagesInventory
-      .where(location_id: location_id)
-      .group(:package_id)
-      .having('SUM(quantity) > 0')
-      .pluck(:package_id)
-
-    Stocktake.watcher_off do
-      # We disable the watch module during this operation to
-      # avoid automatically recomputing counter caches on revision creation
-
-      ActiveRecord::Base.transaction do
-        package_ids.map do |pid|
-          StocktakeRevision.find_or_create_by(package_id: pid, stocktake_id: id) do |revision|
-            revision.quantity       = 0
-            revision.dirty          = true
-            revision.state          = 'pending'
-            revision.created_by_id  = User.current_user&.id || User.system_user.id
-          end
-        end
-
-        compute_counters!
-      end
-    end
+    attrs = {
+      created_by_id: User.current_user&.id || User.system_user.id,
+      state: "'pending'",
+      dirty: true,
+      stocktake_id: id,
+      quantity: 0,
+    }
+
+    ActiveRecord::Base.connection.execute <<-SQL
+      INSERT INTO stocktake_revisions (package_id, created_at, updated_at, #{attrs.keys.join(',')})
```

Comment:
Whilst this code is currently safe, can further protect with `sanitize_sql` to protect from future accidental inclusions of user unsanitized input.